mantidproject/mantid
# New Algorithm: Abins2D

**Description of work.**

Add a new Abins2D algorithm. This shares a lot of code with the existing Abins algorithm, but provides a different front-end and returns 2-D outputs. These are suitable for a different set of instruments with different user parameters; as there are no immediate plans to develop a complex Abins Interface, this split over two algorithms seems the best bet for providing a reasonable user interface in each case.

There have been a few previous iterations of this PR. This one is based on a rework of the underlying maths to efficiently simulate spectra in 2-D q-energy space.

Several ISIS instruments are supported, making use of PyChop to compute resolution functions and identify angle limits. Rather than use pychop for every point, a regular set of points is computed and a polynomial fitted to obtain width (sigma) as a function of energy. This is then passed to the instrumental broadening algorithm which uses linear interpolation to reduce the number of Gaussian kernel evaluations.

Additionally, the ILL-PANTHER instrument is included. The resolution-E_i-Delta_E relationship is fitted to data provided by ILL.

**Todo:**
- [x] rename TwoDMap to Ideal2D
- [x] add systemtest
- [x] get doctests working

**To test:**
Unit tests, systemtest and doctests are provided.

I would also request a manual test if possible; open Abins2D, play with parameters, load a .phonon file. It should support the same input files as Abins.

**Release notes:**
This is a substantial, long-awaited new feature.
A release note has been added under Direct, as this algorithm is for simulation of direct-geometry instruments. Note that previous Abins notes have always been classified as "Indirect"; this raises questions about how best to classify Abins developments going forward.

- [x] Add a pretty picture to release note

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
